File: TouchNothing/Services/SupabaseConfig.swift
import Foundation

enum SupabaseConfigError: Error {
    case missingSecretsFile
    case invalidURL
    case missingAnonKey
}

enum SupabaseConfig {
    static func load() throws -> (url: URL, anonKey: String) {
        guard let secretsPath = Bundle.main.path(forResource: "SupabaseSecrets", ofType: "plist") else {
            throw SupabaseConfigError.missingSecretsFile
        }

        guard let secretsDictionary = NSDictionary(contentsOfFile: secretsPath) as? [String: Any] else {
            throw SupabaseConfigError.missingSecretsFile
        }

        guard let urlString = secretsDictionary["SUPABASE_URL"] as? String,
              let projectURL = URL(string: urlString),
              !urlString.contains("YOUR_PROJECT_REF") else {
            throw SupabaseConfigError.invalidURL
        }

        guard let anonKey = secretsDictionary["SUPABASE_ANON_KEY"] as? String,
              !anonKey.contains("YOUR_ANON_KEY") else {
            throw SupabaseConfigError.missingAnonKey
        }

        return (projectURL, anonKey)
    }
}

File: app/src/main/java/com/touchnothing/app/data/remote/SupabaseConfig.kt
package com.touchnothing.app.data.remote

import com.touchnothing.app.BuildConfig

object SupabaseConfig {
    val url: String?
        get() {
            val value = BuildConfig.SUPABASE_URL
            if (value.isBlank() || value.contains("YOUR_PROJECT_REF")) return null
            return value.trimEnd('/')
        }

    val anonKey: String?
        get() {
            val value = BuildConfig.SUPABASE_ANON_KEY
            if (value.isBlank() || value.contains("YOUR_ANON_KEY")) return null
            return value
        }

    val isConfigured: Boolean
        get() = url != null && anonKey != null
}
